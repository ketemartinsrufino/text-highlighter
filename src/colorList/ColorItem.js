import { useState } from 'react';
import styled from 'styled-components';

const ColorItemStyled = styled.div`
    border: ${({isSelected}) => isSelected ? 2: 1}px black solid;
    height:30px;
    width: 50px;
    background-color: ${({color}) => color};

    :hover {
        cursor: pointer:
    }
`;

const ColorItem = ({color}) => {

    return <ColorItemStyled 
            color="red" 
            isSelected={this.isSelected("red")}
            onClick={this.getOnClick("red")}
        />
}