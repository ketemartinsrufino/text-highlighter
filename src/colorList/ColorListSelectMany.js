import React, { Component } from 'react';
import ColorList from './ColorList';
import PropTypes from 'prop-types';

class ColorListSelectMany extends Component {
  constructor(props) {
    super(props);
    this.state = {
        selectedColors: []
    };
  }

  selectColor = (color) => {
      this.setState((state) => {
        const selectedColors = [...state.selectedColors, color];
        this.props.setSelectedColors(selectedColors);
        
        console.log('selectColor', {selectedColors: selectedColors})

        return {
          selectedColors
        }
      });
  }

  removeColor = (index) => {
    this.setState((state) => {
      const colors = [...state.selectedColors];
      colors.splice(index, 1);
      this.props.setSelectedColors(colors);

      return ({
        selectedColors: colors
      }
    )});
  };

  onClick = (color) => {
    const colors = this.state.selectedColors;
    const index = colors.indexOf(color);
    if(!colors.length || index === -1) {
      this.selectColor(color)
    } else {
      this.removeColor(index);
    }
  }
    
    render(){        
      return (
        <ColorList
          selectedColors={this.state.selectedColors}
          onClick={this.onClick}
        />
      )
    }
}

ColorListSelectMany.propTypes = {
  setSelectedColors: PropTypes.func.isRequired
};

export default ColorListSelectMany;