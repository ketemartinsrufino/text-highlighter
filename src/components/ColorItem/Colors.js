export const COLORS = {
  red: '#ef5350', 
  yellow: '#ffeb3b', 
  green: '#8eda8e'
}