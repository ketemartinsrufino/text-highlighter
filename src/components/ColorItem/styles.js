import styled from 'styled-components';

export const ColorItemStyled = styled.div`
  background-color: ${({color}) => color};
  border-radius: 50%;
  height: 30px;
  width: 30px;
  margin: 0 5px;
  border: 1px #a97c7c solid;

  ${({ isSelected }) => isSelected && `
    border: 2px black solid
  `}

  :hover {
    cursor: pointer:
  }
`;