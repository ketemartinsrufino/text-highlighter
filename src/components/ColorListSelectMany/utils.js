
const addColor = (color, colors=[]) => color ? [...colors, color] : colors;

const removeColor = (color, colors=[]) => {
  const newColors = [...colors];
  const index = newColors.indexOf(color);
  newColors.splice(index, 1);
  return newColors;
};

export const handleSelectColor = (color, colors=[]) => {
  if(!colors.length || !colors.includes(color)) {
   return addColor(color, colors)
  } else {
    return removeColor(color, colors);
  }
}

