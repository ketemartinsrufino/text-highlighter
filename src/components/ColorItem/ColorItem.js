import React from 'react';
import PropTypes from 'prop-types';
import { ColorItemStyled } from './styles';

const ColorItem = ({ color, isSelected, onClick }) => (
  <ColorItemStyled
    color={color}
    isSelected={isSelected}
    onClick={() => onClick(color)}
    className={`ColorItem${isSelected ? ' selected' : ''}`}
  />
)

ColorItem.defaultProps = {
  isSelected: false
}

ColorItem.propTypes = {
  isSelected: PropTypes.bool,
  color: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

export default ColorItem;