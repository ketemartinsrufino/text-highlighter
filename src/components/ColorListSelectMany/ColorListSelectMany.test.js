import React from 'react';
import { shallow, mount } from 'enzyme';
import ColorListSelectMany from './ColorListSelectMany';
import toJson from 'enzyme-to-json';
import { COLORS } from '../ColorItem/Colors';

describe('ColorListSelectMany', () => {
  it('is defined with a default value', () => {
    const component = shallow(
      <ColorListSelectMany setSelectedColors={jest.fn()}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('receive a list of "selectedColors"', () => {
    const component = shallow(
      <ColorListSelectMany 
        selectedColors={[COLORS.yellow, COLORS.red]} 
        setSelectedColors={jest.fn()}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  });

});