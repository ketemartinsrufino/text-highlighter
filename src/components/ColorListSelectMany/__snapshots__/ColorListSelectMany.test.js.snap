// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorListSelectMany is defined with a default value 1`] = `
<ColorList
  colors={
    Array [
      "#ef5350",
      "#ffeb3b",
      "#8eda8e",
    ]
  }
  onClick={[Function]}
  selectedColors={Array []}
/>
`;

exports[`ColorListSelectMany receive a list of "selectedColors" 1`] = `
<ColorList
  colors={
    Array [
      "#ef5350",
      "#ffeb3b",
      "#8eda8e",
    ]
  }
  onClick={[Function]}
  selectedColors={
    Array [
      "#ffeb3b",
      "#ef5350",
    ]
  }
/>
`;
